src/Microsoft.Data/DataFrame.IO.cs

```diff
@@ -0,0 +1,261 @@
+                                string[] names = null, Type[] dtypes = null,
+                                int nrows = -1, int guess_rows = 10,
+                                Encoding encoding = null, bool index = false)
+        {
+            return ReadStream(() => new StreamReader(filename, encoding ?? Encoding.ASCII),
+                              sep: sep, header: header, names: names, dtypes: dtypes, nrows: nrows,
+                              guess_rows: guess_rows, index: index);
+        }
+
+        private delegate StreamReader FunctionCreateStreamReader();
+
+        /// <summary>
+        /// Reads a text file as a IDataView.
+        /// Follows pandas API.
+        /// </summary>
+        /// <param name="createStream">function which creates a stream</param>
+        /// <param name="sep">column separator</param>
+        /// <param name="header">has a header or not</param>
+        /// <param name="names">column names (can be empty)</param>
+        /// <param name="dtypes">column types (can be empty)</param>
+        /// <param name="nrows">number of rows to read</param>
+        /// <param name="guess_rows">number of rows used to guess types</param>
+        /// <param name="index">add one column with the row index</param>
+        /// <returns>DataFrame</returns>
+        private static DataFrame ReadStream(FunctionCreateStreamReader createStream,
+                                char sep = ',', bool header = true,
+                                string[] names = null, Type[] dtypes = null,
+                                long nrows = -1, int guess_rows = 10, bool index = false)
+        {
+            var lines = new List<string[]>();
+            long rowline = 0;
+
+            // First pass: schema and number of rows.
+            using (var st = createStream())
+            {
+                string line = st.ReadLine();
+                int nbline = 0;
+                while (line != null && (nrows == -1 || rowline < nrows))
+                {
+                    var spl = line.Split(sep);
+                    if (header && nbline == 0)
+                    {
+                        if (names == null)
+                            names = spl;
+                    }
+                    else
+                    {
+                        ++rowline;
+                        if (lines.Count < guess_rows)
+                            lines.Add(spl);
+                    }
+                    ++nbline;
+                    line = st.ReadLine();
+                }
+            }
+
+            if (lines.Count == 0)
+                throw new FormatException(Strings.EmptyFile);
+            int numCol = lines.Select(c => c.Length).Max();
+            var df = new DataFrame();
+
+            // Guesses types and adds columns.
+            for (int i = 0; i < numCol; ++i)
+            {
+                Type kind = GuessKind(i, lines);
+                switch (kind)
```

Comment:
Support exists for now to create the different types of columns, but `GuessKind` itself only returns one of `string, bool, float` at the moment.